robosuite/robots/single_arm.py

```diff
@@ -175,11 +175,13 @@ def reset(self, deterministic=False):
         # First, run the superclass method to reset the position and controller
         super().reset(deterministic)
 
-        if not deterministic:
-            # Now, reset the gripper if necessary
-            if self.has_gripper:
+        # Now, reset the gripper if necessary
+        if self.has_gripper:
+            if not deterministic:
                 self.sim.data.qpos[self._ref_gripper_joint_pos_indexes] = self.gripper.init_qpos
 
+            self.gripper.current_action = np.zeros(self.gripper.dof)
```

Comment:
does this need to happen if we're not using a deterministic reset? it seems like this will already happen if hard resets are enabled (as issue #386 mentions, the problem is only there when not using hard resets)

Reply:
As the docstring of the method refers to hard resets from XML files as only an example for deterministic resets I thought implementing it like this would make it more clear. However, in practice, `deterministic = True` should exclusively be the case when loading from XML files, is that correct?

Reply:
I see your point - it might be good to refresh it regardless